File: Java Fundamentals/BasicSyntaxCondStatementsLoops/src/Vacation.java
import java.util.Scanner;

public class Vacation {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int numOfPeople = Integer.parseInt(scan.nextLine());
        String groupType = scan.nextLine();
        String dayOfWeek = scan.nextLine();
        double pricePerNight = 0;
        double discountPriceAdjustment = 1;
        double totalPrice = 0;

        //calc pricePerNight
        switch (groupType){
            case "Students":
                switch (dayOfWeek){
                    case "Friday": pricePerNight = 8.45; break;
                    case "Saturday": pricePerNight = 9.8; break;
                    case "Sunday": pricePerNight = 10.46; break;
                }
                break;
            case "Business":
                switch (dayOfWeek){
                    case "Friday": pricePerNight = 10.9; break;
                    case "Saturday": pricePerNight = 15.60; break;
                    case "Sunday": pricePerNight = 16; break;
                }
                break;
            case "Regular":
                switch (dayOfWeek){
                    case "Friday": pricePerNight = 15; break;
                    case "Saturday": pricePerNight = 20; break;
                    case "Sunday": pricePerNight = 22.5; break;
                }
                break;
        }

        //calc discountPriceAdjustment
        if(groupType.equals("Students") && numOfPeople >= 30){
            discountPriceAdjustment -= 0.15;
        } else if (groupType.equals("Business") && numOfPeople >= 100) {
            numOfPeople -= 10;
        } else if (groupType.equals("Regular") && numOfPeople >= 10 && numOfPeople <= 20) {
            discountPriceAdjustment -= 0.05;
        }
        totalPrice = numOfPeople * pricePerNight * discountPriceAdjustment;

        System.out.printf("Total price: %.2f", totalPrice);

    }
}

File: JS Front-End/JS Syntax Fundamentals/Vacation.js
function printTotal(groupSize, groupType, dayOfWeek){
    function getDayIndex(day){
        if(day == "Friday"){
            return 0;
        }else if(day == "Saturday"){
            return 1;
        }else if(day == "Sunday"){
            return 2;
        }
    }

    let studentPrices = [8.45, 9.8, 10.46];
    let businessPrices = [10.9, 15.6, 16];
    let regularPrices = [15, 20, 22.5];
    let dayIndex = getDayIndex(dayOfWeek);
    let price = 0;
    switch(groupType){
        case "Students": price += studentPrices[dayIndex]; break;
        case "Business": price += businessPrices[dayIndex]; break; 
        case "Regular": price += regularPrices[dayIndex]; break;
    }

    price *= groupSize;

    if(groupType == "Students" && groupSize >= 30){
        price *= 0.85;
    }else if(groupType == "Business" && groupSize >= 100){
        price -= (price / groupSize * 10);
    }else if(groupType == "Regular" && groupSize >= 10 && groupSize <= 20){
        price *= 0.95;
    }

    console.log(`Total price: ${price.toFixed(2)}`);
}

printTotal(30,"Students","Sunday");
printTotal(40, "Regular","Saturday");